interop/tapir/src/main/scala/caliban/interop/tapir/HttpUploadAdapter.scala

```diff
@@ -0,0 +1,132 @@
+package caliban.interop.tapir
+
+import caliban._
+import caliban.interop.tapir.TapirAdapter._
+import caliban.uploads.{ FileMeta, GraphQLUploadRequest, Uploads }
+import sttp.model.{ headers => _, _ }
+import sttp.tapir.Codec.JsonCodec
+import sttp.tapir._
+import sttp.tapir.model.ServerRequest
+import sttp.tapir.server.ServerEndpoint
+import zio._
+
+import java.nio.charset.StandardCharsets
+import scala.util.Try
+
+sealed trait HttpUploadAdapter[-R, E] { self =>
```

Comment:
Isn't `E` covariant? Not sure if it hurts anything for it not to be here.

Reply:
I couldn't make this work because it's used in both covariant and contravariant inside the implementation. But I think it doesn't matter since it's used only in the constructor.